init.lua

```diff
@@ -489,63 +454,63 @@ end
 
 local function get_item_usages(item)
 	local usages = {}
-	for name, def in pairs(reg_items) do
-		if not (def.groups.not_in_craft_guide == 1 or
-				def.groups.not_in_creative_inventory == 1) and
-				get_recipe(name).items and
-				def.description and def.description ~= "" then
-
-			local recipes = get_recipes(name)
-			for i = 1, #recipes do
-				local recipe = recipes[i]
-				if item_in_recipe(item, recipe) then
-					usages[#usages + 1] = recipe
-				end
+
+	for _, recipes in pairs(recipes_cache) do
+		for i = 1, #recipes do
+			local recipe = recipes[i]
+			if item_in_recipe(item, recipe) then
+				usages[#usages + 1] = recipe
 			end
 		end
 	end
 
+	if fuel_cache[item] then
+		usages[#usages + 1] = {type="fuel", width=1, items={item}}
+	end
+
 	return usages
 end
 
 local function get_inv_items(player)
-	local invlist = player:get_inventory():get_list("main")
+	local lists = player:get_inventory():get_lists()
 	local inv_items = {}
 
-	for i = 1, #invlist do
-		local stack = invlist[i]
-		if not stack:is_empty() then
-			inv_items[#inv_items + 1] = stack:get_name()
+	for _, list in pairs(lists) do
+		for i = 1, #list do
+			local stack = list[i]
+			if not stack:is_empty() then
+				local name = stack:get_name()
+				if not inv_items[name] and reg_items[name] then
+					inv_items[#inv_items + 1] = name
+				end
+			end
 		end
 	end
 
 	return inv_items
 end
 
-local function progressive_show_recipe(recipe, inv_items)
-	local inv_items_size = #inv_items
-	for _, item in pairs(recipe.items) do
-		local item_in_inv
-		if item:sub(1,6) == "group:" then
-			local groups = extract_groups(item)
-			for i = 1, inv_items_size do
-				local item_def = reg_items[inv_items[i]]
-				if item_def then
-					local item_groups = item_def.groups
-					if item_has_groups(item_groups, groups) then
-						item_in_inv = true
-					end
-				end
+local function item_in_inv(item, inv_items)
+	if item:sub(1,6) == "group:" then
+		local groups = extract_groups(item)
+		for i = 1, #inv_items do
+			local item_groups = reg_items[inv_items[i]].groups
+			if item_has_groups(item_groups, groups) then
+				return true
 			end
-		else
-			for i = 1, inv_items_size do
-				if inv_items[i] == item then
-					item_in_inv = true
-				end
+		end
+	else
+		for i = 1, #inv_items do
```

Comment:
Localize `#inv_items`

Reply:
It's in a separate function now so would require an additional parameter. Also I really think that would not improve performance in any measurable way.

Reply:
Didn't see it's in a separate function, nevermind then.